rbi/stdlib/date.rbi

```diff
@@ -947,8 +947,8 @@ class Date
   # Date.parse('20010203')            #=> #<Date: 2001-02-03 ...>
   # Date.parse('3rd Feb 2001')        #=> #<Date: 2001-02-03 ...>
   # ```
-  sig {params(arg0: T.untyped).returns(T.attached_class)}
-  def self.parse(*arg0); end
+  sig {params(string: String, comp: T::Boolean, state: T.any(Integer, Float)).returns(T.attached_class)}
+  def self.parse(string='-4712-01-01', comp=true, state=Date::ITALY); end
```

Comment:
One specific thing I'm not sure about is whether to type `comp` as `T::Boolean` here. The _intention_ is that it's a boolean—it defaults to `true` and can be `false`—but Ruby being Ruby means that technically any value is fine here as they'll auto-coerce to booleans. I'd be willing to relax this to `T.untyped` if we want, but I have it as `T::Boolean` to surface places where incorrect assumptions are being made about the code.